File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(seqA, seqB) {
  if (seqA.length != seqB.length) {
    throw "DNA strands must be of equal length.";
  }
  var length = seqA.length;
  var diff = 0;
  for (var i = 0; i < length; i++) {
    if (seqA[i] != seqB[i]) {
      diff++;
    }
  }
  return diff;
}

module.exports = Hamming;

File: c/hamming/hamming.c
#include "hamming.h"

int compute(const char *lhs, const char *rhs)
{
    int lsize = strlen(lhs);
    int rsize = strlen(rhs);
    if (lsize != rsize) {
        return -1;
    }
    int diff = 0;
    for (int i = 0 ; i <= lsize; i++) {
        if (lhs[i] != rhs[i]) {
            diff++;
        }
    }
    return diff;
}